db/view/view_update_generator.cc

```diff
@@ -216,12 +216,17 @@ future<> view_update_generator::start() {
 }
 
 void view_update_generator::do_abort() noexcept {
-}
+    if (_as.abort_requested()) {
+        return;
+    }
```

Comment:
It seems that if as.abort_requested already, you avoid you won't do things like db.unplug_view_update_generator(). I don't exactly understand the context of calling this function, but itsn't it actually common to call it after abort_requested is already set? (sorry, I don't have a good understanding of the shutdown process).

Reply:
The `do_abort()` can be called two times. First -- when `stop_signal` requests abort on its abort source. In that case the v.u.g.'s subscription calls do_abort(). Second -- when v.u.g. is stopped and its `.stop()` method is called. However, the first invocation may not happen in case start aborts early. In that case only `v.u.g.::stop()` is called and it should do what `do_abort()` does.

Said that -- the `do_abort()` should take care of not aborting v.u.g. for the 2nd time, thus this check. I see that `abort_source::request_abort()` can be called multiple times, so does the `unplug_view_update_generator()`, as well as signalling cond.var. But that's some intimate knowledge, so here I prefer explicitly being on the safe side.

Reply:
Added code comment about it too